File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/local/constants/StationList.java
package com.zuk0.gaijinsmash.riderz.data.local.constants;

import java.util.HashMap;
import java.util.Map;

public class StationList {

    //private ArrayMap<String, String> stationMap;

    public static HashMap<String, String> stationMap = new HashMap<>();

    static {
        stationMap.put("12TH", "12th St. Oakland City Center");
        stationMap.put("16TH", "16th St. Mission");
        stationMap.put("19TH", "19th St. Oakland");
        stationMap.put("24TH", "24th St. Mission");
        stationMap.put("ASHB", "Ashby");
        stationMap.put("ANTC", "Antioch");
        stationMap.put("BALB", "Balboa Park");
        stationMap.put("BAYF", "Bay Fair");
        stationMap.put("CAST", "Castro Valley");
        stationMap.put("CIVC", "Civic Center");
        stationMap.put("COLS", "Coliseum");
        stationMap.put("COLM", "Colma");
        stationMap.put("CONC", "Concord");
        stationMap.put("DALY", "Daly City");
        stationMap.put("DBRK", "Downtown Berkeley");
        stationMap.put("DUBL", "Dublin/Pleasanton");
        stationMap.put("DELN", "El Cerrito del Norte");
        stationMap.put("PLZA", "El Cerrito Plaza");
        stationMap.put("EMBR", "Embarcadero");
        stationMap.put("FRMT", "Fremont");
        stationMap.put("FTVL", "Fruitvale");
        stationMap.put("GLEN", "Glen Park");
        stationMap.put("HAYW", "Hayward");
        stationMap.put("LAFY", "Lafayette");
        stationMap.put("LAKE", "Lake Merrit");
        stationMap.put("MCAR", "MacArthur");
        stationMap.put("MLBR", "Millbrae");
        stationMap.put("MONT", "Montgomery St.");
        stationMap.put("NBRK", "North Berkeley");
        stationMap.put("NCON", "North Concord/Martinez");
        stationMap.put("OAKL", "Oakland Int'l Airport");
        stationMap.put("ORIN", "Orinda");
        stationMap.put("PITT", "Pittsburg/Bay Point");
        stationMap.put("PCTR", "Pittsburg Center");
        stationMap.put("PHIL", "Pleasant Hill");
        stationMap.put("POWL", "Powell St.");
        stationMap.put("RICH", "Richmond");
        stationMap.put("ROCK", "Rockridge");
        stationMap.put("SBRN", "San Bruno");
        stationMap.put("SFIA", "San Francisco International Airport");
        stationMap.put("SANL", "San Leandro");
        stationMap.put("SHAY", "South Hayward");
        stationMap.put("SSAN", "South San Francisco");
        stationMap.put("UCTY", "Union City");
        stationMap.put("WARM", "Warm Springs/South Fremont");
        stationMap.put("WCRK", "Walnut Creek");
        stationMap.put("WDUB", "West Dublin");
        stationMap.put("WOAK", "West Oakland");
    }

    public static String getAbbrFromStationName(String name) {
        String abbr = "";
        for(Map.Entry<String, String> e : StationList.stationMap.entrySet()) {
            if(e.getValue().equalsIgnoreCase(name)) {
                abbr = e.getKey();
            }
        }
        return abbr;
    }

}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/local/constants/StationList.kt
package com.zuk0.gaijinsmash.riderz.data.local.constants

import java.util.*

object StationList {
    //private ArrayMap<String, String> stationMap;
    var stationMap = HashMap<String, String>()

    fun getAbbrFromStationName(name: String?): String {
        var abbr = ""
        for ((key, value) in stationMap) {
            if (value.equals(name, ignoreCase = true)) {
                abbr = key
            }
        }
        return abbr
    }

    init {
        stationMap["12TH"] = "12th St. Oakland City Center"
        stationMap["16TH"] = "16th St. Mission"
        stationMap["19TH"] = "19th St. Oakland"
        stationMap["24TH"] = "24th St. Mission"
        stationMap["ASHB"] = "Ashby"
        stationMap["ANTC"] = "Antioch"
        stationMap["BALB"] = "Balboa Park"
        stationMap["BAYF"] = "Bay Fair"
        stationMap["CAST"] = "Castro Valley"
        stationMap["CIVC"] = "Civic Center"
        stationMap["COLS"] = "Coliseum"
        stationMap["COLM"] = "Colma"
        stationMap["CONC"] = "Concord"
        stationMap["DALY"] = "Daly City"
        stationMap["DBRK"] = "Downtown Berkeley"
        stationMap["DUBL"] = "Dublin/Pleasanton"
        stationMap["DELN"] = "El Cerrito del Norte"
        stationMap["PLZA"] = "El Cerrito Plaza"
        stationMap["EMBR"] = "Embarcadero"
        stationMap["FRMT"] = "Fremont"
        stationMap["FTVL"] = "Fruitvale"
        stationMap["GLEN"] = "Glen Park"
        stationMap["HAYW"] = "Hayward"
        stationMap["LAFY"] = "Lafayette"
        stationMap["LAKE"] = "Lake Merrit"
        stationMap["MCAR"] = "MacArthur"
        stationMap["MLBR"] = "Millbrae"
        stationMap["MONT"] = "Montgomery St."
        stationMap["NBRK"] = "North Berkeley"
        stationMap["NCON"] = "North Concord/Martinez"
        stationMap["OAKL"] = "Oakland Int'l Airport"
        stationMap["ORIN"] = "Orinda"
        stationMap["PITT"] = "Pittsburg/Bay Point"
        stationMap["PCTR"] = "Pittsburg Center"
        stationMap["PHIL"] = "Pleasant Hill"
        stationMap["POWL"] = "Powell St."
        stationMap["RICH"] = "Richmond"
        stationMap["ROCK"] = "Rockridge"
        stationMap["SBRN"] = "San Bruno"
        stationMap["SFIA"] = "San Francisco International Airport"
        stationMap["SANL"] = "San Leandro"
        stationMap["SHAY"] = "South Hayward"
        stationMap["SSAN"] = "South San Francisco"
        stationMap["UCTY"] = "Union City"
        stationMap["WARM"] = "Warm Springs/South Fremont"
        stationMap["WCRK"] = "Walnut Creek"
        stationMap["WDUB"] = "West Dublin"
        stationMap["WOAK"] = "West Oakland"
    }
}